test/cached_results_test.rb

```diff
@@ -0,0 +1,40 @@
+require 'test_helper'
+
+describe AccessControl do
+  it 'caches the result of general permitted action' do
+    actor = User.create!
+    query = AccessControl::Query.new(actor)
+    query.can?(1, Post).must_equal false
+
+    AccessControl::PermittedAction.create!(
+      id: SecureRandom.uuid,
+      actor: actor,
```

Comment:
@armilam or @rreinhardt9 
Do we need to consider multiple actors in the test suite?

In the `can?` method it looks like we cache use the actor's class name in the cache - but our `where` query only passes in the `@actor` reference.  Is ActiveRecord doing something magically with `actor_type` or do our `where` queries need to specify that class name explicitly like
```
       PermittedActionOnObject.where(
          actor: @actor,
          actor_type: @actor.class.name,
          action: action_id,
          object_type: String(object_type),
          object_id: object_id
        ).exists?
```

The same question for `PermittedActionsOnObject::can?`

Reply:
How should we consider the use of `segment_id` in the API?

Do we initialize AccessControl::Query with an `actor` and `segment_id`?